lib/plugins/exchange/cctx/cctx.js

```diff
@@ -0,0 +1,57 @@
+var ccxt = require('ccxt')
+const coinUtils = require('../../../coin-utils')
+const _ = require('lodash/fp')
+const common = require('../../common/cctx')
+
+function trade (side, account, cryptoAtoms, fiatCode, cryptoCode, exchangeName) {
+  try {
+    const exchange = setUpExchange(account, exchangeName)
+    const symbol = common.verifyCurrencies(exchangeName, fiatCode, cryptoCode)
+    const amount = coinUtils.toUnit(cryptoAtoms, cryptoCode).toFixed(8)
+    if (exchangeName === 'itbit') {
+      return exchange.fetchOrderBook(symbol)
+        .then(orderBook => {
+          return exchange.createOrder(symbol, 'limit', side, amount, calculatePrice(side, amount, orderBook), { walletId: account.walletId })
+        })
+    }
+    return exchange.createOrder(symbol, 'market', side, amount)
+  } catch (e) {
+    return Promise.reject(e)
+  }
+}
+
+function setUpExchange (account, exchangeName) {
+  // map given credentials to cctx properties
+  if (!_.includes(exchangeName, ccxt.exchanges)) {
+    throw new Error(`Exchange ${exchangeName} not supported by ccxt.`)
+  }
+
+  switch (exchangeName) {
+    case 'itbit':
+      if (!account.clientKey || !account.clientSecret || !account.userId || !account.walletId)
+        throw new Error('Must provide user ID, wallet ID, client key, and client secret')
+      return new ccxt[exchangeName](_.mapKeys((key) => { return key === 'clientKey' ? 'apiKey' : key === 'clientSecret' ? 'secret' : key === 'userId' ? 'uid' : key }, _.omit(['walletId'], account)))

```

Comment:
If we move the ternary operators to a const it becomes more readable
`
const someName = key === 'clientKey' ? 'apiKey' : key === 'clientSecret' ? 'secret' : key === 'userId' ? 'uid' : key
`
`
_.mapKeys((key) => someName
`

same for the others

Reply:
This one should have brackets IMO. Also there's too much code a single line, try to stick to 80 col length.